### PR TITLE
Scale test upgrade

### PR DESCRIPTION
**Pull request by KrystianMarek**
#### What's this change do?
- Added load test
- Added test exposing a bug in result verification
- Various improvements
- PyTest flags changed
- ToDo updated
- Scale test is longer runs by default

#### Where should the reviewer start?
tests/functional/*

### CODE REVIEW

File: tests/functional/test_900_BIP_scale.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            try:
+                bip_client.deploy_app_service(payload)
+                bip_client.verify_deployment_result(
+                    payload, test_run_log_dir)
+                test_results[strip_payload_name(
+                    payload['name'])]['completed'] = deployment_no + 1
+
+            except (AppServiceDeploymentException, RESTException,
+                    AppServiceDeploymentVerificationException) as ex:
+                test_results[strip_payload_name(payload['name'])].setdefault(
```
Comment (RafalKorepta) [inline tests/functional/test_900_BIP_scale.py]: test_results aren't collected at the end of scale run.

File: tests/functional/test_900_BIP_scale.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+    log_dir = os.path.join("logs", get_config['session_id'], 'run')
+    mk_dir(log_dir)
+
+    for payload_file in get_payload_files(get_config):
+
+        payload_basename = get_payload_basename(payload_file)
+        if payload_basename in [
+            'test_vs_standard_https_multi_listeners',
+            'test_vs_standard_http_ipv6'
```
Comment (RafalKorepta) [inline tests/functional/test_900_BIP_scale.py]: There is problem with test_pools_fixup_4 on BIG-IP 12 and up. Comment it or add payload_basename to exclude from scale run.

File: tests/appservices/test_TestTools.py [hunk 3]
```diff
@@ -14,28 +14,134 @@
...
-
-    assert len(payloads) == 5
 
     # I can not wait for the day
     #  when we remove this 'payload dependency mechanism'
-    assert len(dependants.keys()) == 10
+    assert len(dependants.keys()) >= 8
+
+    assert get_payload_dependencies(dependants, 'test_pools') == [
+        {
```
Comment (caphrim007) [inline tests/appservices/test_TestTools.py]: I would change your assertions here to not directly assert dictionary order unless you are deliberately using OrderedDict. Dicts in Python are inherently unordered and so this test can technically return `False` in the event that Python decides to order them according to its own internal performance calculations.

The better thing to do would be to assert that there is a length to that list and that the list includes those items and keys.

For example.

foo = get_payload_dependencies(dependants, 'test_pools'))

assert len(foo) == 10
assert 'delete_override' in foo[0]
assert foo[0]['delete_override'] is True

Lists _are_ ordered, so you dont need to worry about the 0'th index not being the first item you specified there. This method is more verbose, but it's also more correct.

File: tests/functional/test_299_load.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    Deployment of the iApp fails randomly with:
+    - BIP freezes on current_time=1500831066 result=
+    - BIP freezes on result=DEFERRED_CMDS_IN_PROGRESS
+    - 01070734:3: Configuration error: Invalid mcpd context, folder not found
+
+    Expectation:
+    All Application Services are deployed without any issues
+
+    This test focuses on iStat issues, test is marked as failed after
+    first exception was thrown.
```
Comment (RafalKorepta) [inline tests/functional/test_299_load.py]: Describe what steps thouse threads are doing. How many of them are running in parallel and how much payloads they are creating. Write what payload is used in the tests description. The re-produce setps aren't clear here.

**Pull request merged**